scripts/rescript_format.js

```diff
@@ -112,9 +112,10 @@ function main(argv, rescript_exe, bsc_exe) {
       if (isSupportedStd(use_stdin)) {
         var crypto = require("crypto");
         var os = require("os");
+        var extension = formattedStdExtensions.includes(use_stdin) ? use_stdin : path.extname(use_stdin)
         var filename = path.join(
           os.tmpdir(),
-          "rescript_" + crypto.randomBytes(8).toString("hex") + use_stdin
+          "rescript_" + crypto.randomBytes(8).toString("hex") + extension
```

Comment:
there are some misunderstading about the CLI.
we only accept `rescript format -stdin .res`

Reply:
We did not show a usage error when passing `foo.res` or `foo/bar.res`, but the latter broke: https://github.com/rescript-lang/rescript-compiler/issues/5131#issuecomment-857045941

If you're OK with the behavior change (`foo.res` will no longer work), I'll change it to only accept extensions.

Reply:
Hi, It is just I don't see the point of `format -stdin foo/bar.res`? The name was only used for troubleshooting

Reply:
We accidentally accepted more than just `.res`: we also accepted `foo.res` (which works) and `foo/bar.res` (which crashed).

I also don't see the point, but instead of giving an error when calling `rescript format -stdin foo/bar.res`, the command crashed because we tried to write to `/tmp/rescript_aabbccddeefff1122foo/bar.res`, and the directory `/tmp/rescript_aabbccddeefff1122foo` doesn't exist. Giving a usage error is much better than crashing.

I'll change the behavior to the documented behavior, which is only accepting `.res`.

Reply:
Your explanation is much helpful. We can do a sanity check to avoid file path here?

Reply:
My commit and your comment had a race condition; this is the change I made: https://github.com/rescript-lang/rescript-compiler/pull/5194/commits/5e49fbc2a7baf568490ea1cf09e16effeaafd21c

This makes `rescript format -stdin foo.res` and `rescript format -stdin foo/bar.res` exit with a usage error; is that OK?